v-next/hardhat/test/internal/builtin-plugins/console/task-action.ts

```diff
@@ -0,0 +1,191 @@
+          options,
+        },
+        hre,
+      );
+      assert.equal(replServer.lastError, undefined);
+    });
+
+    it("should throw inside the console if the script throws", async function () {
+      const replServer = await consoleAction(
+        {
+          commands: ['await import("./scripts/throws.js");', ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      ensureError(replServer.lastError);
+    });
+  });
+
+  describe("typescript", function () {
+    useFixtureProject("run-ts-script");
+
+    it("should throw inside the console if script does not exist", async function () {
+      const replServer = await consoleAction(
+        {
+          commands: ['await import("./scripts/non-existent.ts");', ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      ensureError(replServer.lastError);
+    });
+
+    it("should run a script inside the console successfully", async function () {
+      const replServer = await consoleAction(
+        {
+          commands: ['await import("./scripts/success.ts");', ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      assert.equal(replServer.lastError, undefined);
+    });
+
+    it("should throw inside the console if the script throws", async function () {
+      const replServer = await consoleAction(
+        {
+          commands: ['await import("./scripts/throws.ts");', ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      ensureError(replServer.lastError);
+    });
+  });
+
+  describe("context", function () {
+    it("should expose the Hardhat Runtime Environment", async function () {
+      const replServer = await consoleAction(
+        {
+          commands: ["console.log(hre);", ".exit"],
+          history: "",
+          noCompile: false,
+          options,
+        },
+        hre,
+      );
+      assert.equal(replServer.lastError, undefined);
+    });
+  });
+
+  describe("history", function () {
+    let cacheDir: string;
+    let history: string;
+
+    beforeEach(async function () {
+      // We use a temporary cache dir to avoid conflicts with other tests
+      // and global user settings.
+      cacheDir = await fsPromises.mkdtemp(
+        path.resolve(os.tmpdir(), "console-action-test-"),
+      );
+      history = path.resolve(cacheDir, "console-history.txt");
+    });
+
+    afterEach(async function () {
+      // We try to remove the temporary cache dir after each test, but we don't
+      // fail the test if it fails. For example, we have observed that in GHA
+      // on Windows, the temp dir cannot be removed due to permission issues.
+      try {
+        await remove(cacheDir);
+      } catch (error) {
+        log("Failed to remove temporary cache dir", error);
+      }
```

Comment:
I added a try-catch and a comment about an issue I observed in https://github.com/NomicFoundation/hardhat/actions/runs/10332073748/job/28602920453, for example. Have you seen something similar before? Do you think it's worth ensuring we can always delete the temp dirs in CI? In this particular case I think it might have something to do that the temp dir is on `C:` while the runner is set up to have perms and execute in the context of drive `D:`.

Reply:
that's actually sirprising, but I think we can live with it.

Maybe we should move it to the test utils package so that we don't hit this edge case again.

Reply:
I created an issue for this - https://github.com/NomicFoundation/hardhat/issues/5602 If I cannot resolve it, I'm going to add the utility function for this.